.github/workflows/test.yml

```diff
@@ -1,17 +1,19 @@
 name: CI
 
-on: [push]
+on:
+    push:
+        branches:
+            - !master
```

Comment:
There should be no need to run this action on `master` since everything should be built and tested before merge anyway. Alternatively we can keep this, and remove the build step from the deploy action.